Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -172,13 +260,19 @@ private extension TransactionPoster {
                            result: Result<CustomerInfo, BackendError>,
                            subscriberAttributes: SubscriberAttribute.Dictionary?,
                            completion: @escaping CustomerAPI.CustomerInfoResponseHandler) {
-        self.operationDispatcher.dispatchOnMainActor {
+        self.operationDispatcher.dispatchOnMainActor { [cache = self.cache] in
             switch result {
             case let .success(customerInfo):
                 if customerInfo.isComputedOffline {
                     completion(result)
                 } else {
                     self.finishTransactionIfNeeded(transaction) {
+                        Logger.debug(Strings.purchase.transaction_poster_storing_posted_transaction(
+                            productID: transaction.productIdentifier,
+                            transactionID: transaction.transactionIdentifier
+                        ))
+
+                        cache.savePostedTransaction(transaction)
```

Comment:
we don't need to wait for `finishTransactionIfNeeded` to save the transaction, right? Also, we'd have to wait for the main thread? 
I think we should be able to mark it as posted before finishing. In a worst-case scenario where the app crashes before finishing, then we still finish when we see that it has already been posted

Reply:
> we don't need to wait for finishTransactionIfNeeded to save the transaction, right? 
> I think we should be able to mark it as posted before finishing. In a worst-case scenario where the app crashes before finishing, then we still finish when we see that it has already been posted

Yeah I can reorder this.

> ? Also, we'd have to wait for the main thread?
Can you explain?

Reply:
Addressed, this no longer does it in the main thread.